python/benchmarks/basic_functions.py

```diff
@@ -78,20 +77,87 @@ def get_fresh_lib(self):
         self.ac.delete_library("fresh_lib")
         return self.ac.create_library("fresh_lib")
 
-    def time_write(self, rows, num_symbols):
-        for sym in range(num_symbols):
-            self.fresh_lib.write(f"{sym}_sym", self.df)
+    def time_write(self, rows):
+        self.fresh_lib.write(f"sym", self.df)
 
-    def peakmem_write(self, rows, num_symbols):
-        for sym in range(num_symbols):
-            self.fresh_lib.write(f"{sym}_sym", self.df)
+    def peakmem_write(self, rows):
+        self.fresh_lib.write(f"sym", self.df)
 
-    def time_write_short_wide(self, rows, num_symbols):
+    def time_write_short_wide(self, rows):
         self.fresh_lib.write("short_wide_sym", self.df_short_wide)
 
-    def peakmem_write_short_wide(self, rows, num_symbols):
+    def peakmem_write_short_wide(self, rows):
         self.fresh_lib.write("short_wide_sym", self.df_short_wide)
 
+    def time_read(self, rows):
+        self.lib.read(f"sym").data
+
+    def peakmem_read(self, rows):
+        self.lib.read(f"sym").data
+
+    def time_read_short_wide(self, rows):
+        lib = self.ac[get_prewritten_lib_name(BasicFunctions.WIDE_DF_ROWS)]
+        lib.read("short_wide_sym").data
+
+    def peakmem_read_short_wide(self, rows):
+        lib = self.ac[get_prewritten_lib_name(BasicFunctions.WIDE_DF_ROWS)]
+        lib.read("short_wide_sym").data
+
+    def time_read_with_columns(self, rows):
+        COLS = ["value"]
+        self.lib.read(f"sym", columns=COLS).data
+
+    def peakmem_read_with_columns(self, rows):
+        COLS = ["value"]
+        self.lib.read(f"sym", columns=COLS).data
+
+    def time_read_with_date_ranges(self, rows):
+        self.lib.read(f"sym", date_range=BasicFunctions.DATE_RANGE).data
+
+    def peakmem_read_with_date_ranges(self, rows):
+        self.lib.read(f"sym", date_range=BasicFunctions.DATE_RANGE).data
+
+
+class BatchBasicFunctions:
+    number = 5
+    timeout = 6000
+    CONNECTION_STRING = "lmdb://batch_basic_functions?map_size=20GB"
+    DATE_RANGE = pd.date_range("2023-01-01", "2023-01-01")
```

Comment:
I know this hasn't changed, but this date range will filter out all rows, which I'm guessing isn't the intended behaviour

Reply:
Thanks, moved it to `2022-12-31, 2023-01-01`, which should be non empty (last timestamp on all dataframes should be 2023-01-01)